packages/ui/EventCacheable/EventCacheable.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import hoc from '@enact/core/hoc';
+
+const EventCacheable = hoc((config = {defaultEvent: null, type: null}, Wrapped) => {
+	const {defaultEvent, type} = config;
+
+	return class EventParamCacheableBase extends React.Component {
+		cachedEvent = defaultEvent || null
+
+		getEvent = () => (this.cachedEvent)
+
+		setEvent = (event) => {
+			this.cachedEvent = event;
+		}
+
+		eventHandler = (ev) => {
```

Comment:
You'd probably want to forward this on rather than capture the event.

```js
import {forward, handle, call} from '@enact/core/handle'
eventHandler = handle(
  forward(type), // forward to consumers
  call('setEvent'), // invoke this.setEvent(ev)
).bindAs(this, 'eventHandler')
```